docker improvements

casperdcl opened the pull request:
- follow-up to #6
- pin to https://github.com/SyneRBI/SIRF-SuperBuild/commit/e6cc1ff906a184d866c4fe56c8cda79777d392de
- loop: ensure latest tag fetched
- petric: `docker run --init` to forward GHA CI `SIGINT`/`SIGKILL`
- `pin cuda-version 12.8`
  + [ ] upstream: https://github.com/SyneRBI/SIRF-SuperBuild/pull/985
- use [docker stacks conveniences](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html)
  + `NB_GID` (partially reverts 55203a4)
  + startup hooks (`{apt,conda,pip} install`, source sirf env, start `gadgetron`)
    * related to https://github.com/jupyter/docker-stacks/issues/2386
  + [ ] upstream: https://github.com/SyneRBI/SIRF-SuperBuild/pull/985

casperdcl committed:
pin to SIRF 3.9.0

casperdcl committed:
docker NB_GID, minor fixes, startup hooks

- pin cuda-version 12.8
- partially reverts 55203a4
- https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html
  + `NB_GID`
  + startup hooks ({apt,conda,pip} install, source sirf, start gadgetron)

KrisThielemans reviewed (commented):
too much going on here that goes beyond what I know to be able to provide a sensible review. As it works, it's fine for me!

casperdcl committed:
slight configurability increase

casperdcl committed:
bump SIRF

casperdcl committed:
misc tidy

- similar to https://github.com/SyneRBI/PETRIC2/pull/70

casperdcl committed:
ensure latest tag checkout

casperdcl committed:
GHA CI: fix SIGINT/SIGKILL forwarding

- https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-cancellation
- https://docs.docker.com/reference/cli/docker/container/run/#init

casperdcl committed:
minor docker pull
